src/main/java/de/tum/in/www1/artemis/repository/ResultRepository.java

```diff
@@ -162,10 +187,27 @@ SELECT COUNT(r.id)
     @EntityGraph(type = LOAD, attributePaths = { "feedbacks" })
     List<Result> findAllWithEagerFeedbackByAssessorIsNotNullAndParticipation_ExerciseIdAndCompletionDateIsNotNull(Long exerciseId);
 
-    @Query("SELECT COUNT(DISTINCT p) FROM Participation p left join p.results r WHERE p.exercise.id = :exerciseId AND r.assessor IS NOT NULL AND r.assessmentType IN :types AND r.rated = TRUE AND r.completionDate IS NOT NULL AND (p.exercise.dueDate IS NULL OR r.submission.submissionDate <= p.exercise.dueDate)")
+    @Query("""
+            SELECT COUNT(DISTINCT p) FROM Participation p left join p.results r
+            WHERE p.exercise.id = :exerciseId
+                AND r.assessor IS NOT NULL
+                AND r.assessmentType IN :types
+                AND r.rated = TRUE
+                AND r.completionDate IS NOT NULL
+                AND (p.exercise.dueDate IS NULL OR r.submission.submissionDate <= p.exercise.dueDate)
+              """)
     long countNumberOfAssessmentsByTypeForExerciseBeforeDueDate(@Param("exerciseId") Long exerciseId, @Param("types") List<AssessmentType> types);
 
-    @Query("SELECT COUNT(DISTINCT p) FROM Participation p left join p.results r WHERE p.exercise.id = :exerciseId AND r.assessor IS NOT NULL AND r.assessmentType IN :types AND r.rated = FALSE AND r.completionDate IS NOT NULL AND p.exercise.dueDate IS NOT NULL AND r.submission.submissionDate > p.exercise.dueDate")
+    @Query("""
+            SELECT COUNT(DISTINCT p) FROM Participation p left join p.results r
+            WHERE p.exercise.id = :exerciseId
+                AND r.assessor IS NOT NULL
+                AND r.assessmentType IN :types
+                AND r.rated = FALSE
+                AND r.completionDate IS NOT NULL
+                AND p.exercise.dueDate IS NOT NULL
```

Comment:
Is it intended here that exercises without due date do no matter at all?

Reply:
I mean, in this query we want to count assessments after the due date. If there is none it doesn't make sense to count it